pkgs/applications/science/math/caffe/default.nix

```diff
@@ -21,11 +21,17 @@
 }:
 
 let
-  inherit (cudaPackages) cudatoolkit cudnn nccl;
+  inherit (cudaPackages) cudatoolkit nccl;
+  # The default for cudatoolkit 10.1 is CUDNN 8.0.5, the last version to support CUDA 10.1.
+  # However, this caffe does not build with CUDNN 8.x, so we use CUDNN 7.6.5 instead.
+  # Earlier versions of cudatoolkit use pre-8.x CUDNN, so we use the default.
+  cudnn = if lib.versionOlder cudatoolkit.version "10.1"
+    then cudaPackages.cudnn
+    else cudaPackages.cudnn_7_6_5;
 in
 
 assert leveldbSupport -> (leveldb != null && snappy != null);
-assert cudnnSupport -> cudaSupport;
+assert cudnnSupport -> cudaSupport && lib.versionOlder cudnn.version "8.0";
```

Comment:
given this assert, is there any situation in which this package would work with cudnn 7.6.5 as is selected in the let statement above? 

why not just drop support for older cuda/cudnn versions for this package?

Reply:
You're right, the assert there is redundant, I'll remove it.

Per your second question, this package *only* works with older versions of CUDA/CUDNN. Since it won't work with CUDNN 8.x+, it can't run on Ampere or newer.